src/providers/ibmcloud/mod.rs

```diff
@@ -112,6 +129,49 @@ impl IBMGen2Provider {
         }
         output
     }
+
+    // Find the SSH keys in the vendordata file
+    fn fetch_ssh_keys(vendordata: String) -> Result<Vec<String>> {
+        // The ibmcloud vendor data file contains multiple sections, start by finding the section boundary
+        let v: Vec<&str> = vendordata.split("boundary=").collect();
+        let boundary_delimiter = v[1];
```

Comment:
Unfortunately this kind of unchecked indexed access (here and everywhere else) will panic as soon as the input does not match your expectations. Manually parsing data is usually not a good idea.

You may try using a proper mail parser instead, for example https://github.com/staktrace/mailparse, pick up the first entry with a `text/cloud-config` type and pass its content to the YAML deserializer.